### PR TITLE
Refactor requested volumeMounts check for Step and Sidecar Containers

### PR DESCRIPTION
**Pull request by JeromeJu**
This commit refactors the check for the requested volume mounts for sideCar Containers and step Containers to mitigate the possible misleading confusions addressed in #6373.

/kind cleanup
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by Yongxuanzhang — COMMENTED** [review]
/lgtm

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -595,3 +576,20 @@ func usesWindows(tr *v1beta1.TaskRun) bool {
 	osSelector := tr.Spec.PodTemplate.NodeSelector[osSelectorLabel]
 	return osSelector == "windows"
 }
+
+// checkRequestedVolumeMounts checks the existing volume mounts in the current Container and returns a slice
+// of VolumeMounts to be added to the container.
+func checkRequestedVolumeMounts(c corev1.Container, requestedVolumeMounts []corev1.VolumeMount) []corev1.VolumeMount {
```
Comment (jerop) [inline pkg/pod/pod.go:582]: is it right that this function is getting missing volume mounts? it'd be useful to use a precise name for what this function is checking in the requested volumes

```suggestion
// checkRequestedVolumeMounts checks the existing volume mounts in the current Container and returns a slice
// of VolumeMounts to be added to the container.
func getMissingVolumeMounts(c corev1.Container, requestedVolumeMounts []corev1.VolumeMount) []corev1.VolumeMount {
```

**Comment by afrittoli** [comment]
@JeromeJu are you still working on this PR? Do you need reviews?

**Pull request closed without merging**